ide_extensions.md

```diff
@@ -0,0 +1,161 @@
+# IDE Extensions
+## Overview
+An oft-requested feature is an improved `Tiltfile` editing experience in IDEs/editors.
+
+Specifically, for the scope of this spec, we'll consider two high-level aspects:
+ * Syntax highlighting
+ * Advanced IDE features (autocomplete, go to definition, etc.)
+
+Explicitly out of scope for consideration here is any kind of integration with `tilt up` like was done for the [vscode-tilt-status][] extension.
+The focus here is on language-level (Starlark) static analysis.
+
+## Targeted IDEs & Editors
+Our primary target is Visual Studio Code due to its ubiquity and robust extension system.
```

Comment:
i would also mention Monaco for in-browser editing, which is how moment.dev does their autocomplete docs, and will tie into the kinds of interactive stuff i think @nicksieger was interested in

Reply:
Ahh yeah good call, I originally had a mention of this and must have removed it.

The extra nice thing too is that `vscode-languageclient` supports both Node and browsers - in the latter case, it can't spawn processes so you need to handle that but we should be able route the actual RPC comms through websockets.